src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,116 @@
+package io.gingersnapproject.k8s;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec.Builder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@Singleton
+public class CacheRuleInformer {
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleInformer.class);
+
+   @Inject Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   private SharedIndexInformer<ConfigMap> informer;
+
+   void startWatching(@Observes StartupEvent ignore) {
+      if (client.isUnsatisfied() || configuration.configMapName().isEmpty()) {
+         log.info("Kubernetes client not found, not watching config map");
+         return;
+      }
+
+      String ruleName = configuration.configMapName().get();
+      log.info("Informer on rules in {}", ruleName);
+      final long RESYNC_PERIOD = 60 * 1000L;
+      KubernetesClient kc = client.get();
+      informer = kc.configMaps().withName(ruleName).inform(new ConfigMapEventHandler(), RESYNC_PERIOD); 
+      informer.start();
+   }
+
+   public void stop(@Observes ShutdownEvent ignore) {
+      if (informer != null) {
+         log.info("Shutdown informer");
+         informer.close();
+      }
+   }
+
+   private static final class ConfigMapEventHandler implements ResourceEventHandler<ConfigMap> {
+      private interface SendEventFunc{
+         public void sendEvent(String name, EagerCacheRuleSpec rule);
+      }
+      private void processConfigMapAndSend(Set<Map.Entry<String,String>> entries, SendEventFunc f) {
+         for (Entry<String, String> entry : entries) {
+            Builder eRuleBuilder = EagerCacheRuleSpec.newBuilder();
+            try {
+               JsonFormat.parser().ignoringUnknownFields().merge(entry.getValue(), eRuleBuilder);
+            } catch (InvalidProtocolBufferException e) {
+               log.error("Cannot parse eager rule with name {}", entry.getKey());
+               log.debug("{}",e);
```

Comment:
I think the exception needs to be included in the error log so that we can debug these issues if they happen in production.